baseServer_test.go

```diff
@@ -439,6 +433,20 @@ func TestMessagePrint(t *testing.T) {
 	ag := NewTestAgent(server)
 	newMsg := ag.NewTestMessage()
 	newMsg.Print()
+	originalStdout := os.Stdout
+	r, w, _ := os.Pipe()
+	os.Stdout = w
+	newMsg.Print()
+	w.Close()
+	var buf bytes.Buffer
+	io.Copy(&buf, r)
+	output := buf.String()
+	os.Stdout = originalStdout
+	expected := "message received from " + ag.GetID().String() + "\n"
+	if string(output) != string(expected) {
+		t.Error("Expected", expected, "but got", output)
+	}
+
```

Comment:
Sorry to make you do all of this. Is it worth just removing the print method entirely? Surely it defeats the purpose of composition, as we can't really anticipate all of the new fields that the users will add. I suppose we can keep it as overridable, but it may be nicer to just add a print method at the user's request. The base package shouldn't really print anything to console